### PR TITLE
Fix compatibility for latest itrex version

### PR DESCRIPTION
**Pull request by echarlaix**
cc @PenghuiCheng @changwangss

### CODE REVIEW

File: tests/neural_compressor/test_optimization.py [hunk 1]
```diff
@@ -575,3 +497,51 @@ def _compute_metrics(pred):
             self.assertIsInstance(loaded_model_outputs.logits, torch.Tensor)
             # Compare tensor outputs
             # self.assertTrue(torch.allclose(loaded_model_outputs.logits, model_outputs.logits, atol=1e-4))
+
+
+class WeightOnlyQuantizationTest(INCTestMixin):
+    WEIGHT_ONLY_CONFIG = (
+        ("rtn", "int4_clip"),
+        ("rtn", "int8"),
+        # ("gptq", "int4_clip"), # can be enabled after fix in neural-compressor merged
```
Comment (echarlaix) [inline tests/neural_compressor/test_optimization.py:506]: Needs https://github.com/intel/neural-compressor/pull/1729 to be enabled

File: optimum/intel/neural_compressor/quantization.py [hunk 2]
```diff
@@ -178,26 +183,48 @@ def quantize(
...
+        if save_onnx_model and isinstance(self._original_model, ORTModel):
             save_onnx_model = False
             logger.warning("Model provided is an ONNX model, `save_onnx_model` is set to False")
 
         default_name = WEIGHTS_NAME if not isinstance(self._original_model, ORTModel) else ONNX_WEIGHTS_NAME
-        calibration_dataloader = None
         self._set_task()
 
-        if weight_only or not isinstance(quantization_config, PostTrainingQuantConfig):
+        if kwargs.pop("weight_only", None) is None:
```
Comment (PenghuiCheng) [inline optimum/intel/neural_compressor/quantization.py:202]: is it not None?

File: optimum/intel/neural_compressor/quantization.py [hunk 3]
```diff
@@ -207,53 +234,42 @@ def quantize(
...
-                quantization_config = WeightOnlyQuantConfig()
-                algo = "RTN"
-            elif isinstance(quantization_config, WeightOnlyQuantConfig):
-                algo = quantization_config.algorithm
-            else:
-                raise TypeError(
-                    f"For weight-only quantization, `quantization_config` should be an instance of `WeightOnlyQuantConfig`, but got: {type(quantization_config)} instead."
-                )
+            if not isinstance(quantization_config, ITREXQuantizationConfigMixin):
+                raise ValueError("")
```
Comment (PenghuiCheng) [inline optimum/intel/neural_compressor/quantization.py:244]:
```suggestion
                raise ValueError("quantization_config is not an object of ITREXQuantizationConfigMixin, please refer to https://github.com/huggingface/optimum-intel/blob/0c44e0bdc34c64b9713de64c1b4d3695a7a5889d/examples/neural_compressor/language-modeling/run_clm.py#L672")
```

**Review by PenghuiCheng — APPROVED** [review]

File: optimum/intel/neural_compressor/quantization.py [hunk 4]
```diff
@@ -266,45 +282,24 @@ def quantize(
...
                 quantization_config.post_init_xpu()
+
             self._quantized_model = convert_to_quantized_model(
                 self._original_model, quantization_config, device=quantization_config.device
             )
-            # will remove after intel-extension-for-transformers 1.3.3 release.
-            if hasattr(quantization_config, "calib_dataloader"):
-                quantization_config.calib_dataloader = None
+
             self._quantized_model.quantization_config = quantization_config
```
Comment (PenghuiCheng) [inline optimum/intel/neural_compressor/quantization.py:299]:
```suggestion
            self._quantized_model.quantization_config = quantization_config
            self._quantized_model.config.quantization_config = quantization_config
```

File: optimum/intel/neural_compressor/quantization.py (hunk 2 above)
Comment (echarlaix) [reply optimum/intel/neural_compressor/quantization.py:202]: good catch, thanks !

File: optimum/intel/neural_compressor/quantization.py (hunk 4 above)
Comment (PenghuiCheng) [reply optimum/intel/neural_compressor/quantization.py:302]: In the ITREX loading function, 'quantization_config' in 'config' distinguishes between quantized models and those requiring quantization. It is aligned with other quantization methods, like AutoGPTQ, AutoAWQ, and Bitsandbyte.

**Pull request merged**